synctl

```diff
@@ -1894,6 +1894,64 @@ class PatchSyntheticTest(SyntheticTest):
         except:
             print(f"Error: Test {test_id} not found")
 
+    def patch_url(self, url):
+        """update url"""
+        payload = {"configuration": {"url": ""}}
+        if url is not None:
+            payload["configuration"]["url"] = url
+            self.__patch_a_synthetic_test(self.test_id, json.dumps(payload))
+        else:
+            print("url should not be none")
+
+    def patch_follow_redirect(self, follow_redirect):
+        """update follow-redirect"""
+        follow_redirect_options = ["True", "true", "False", "false"]
+        payload = {"configuration": {"followRedirect": ""}}
+        if follow_redirect is not None and follow_redirect in follow_redirect_options:
+            payload["configuration"]["followRedirect"] = follow_redirect
+            self.__patch_a_synthetic_test(self.test_id, json.dumps(payload))
+        else:
+            print("followRedirect should be true/false ")
+
+    def patch_expect_status(self, expect_status):
+        """update expect status"""
+        payload = {"configuration": {"expectStatus": ""}}
+        if expect_status is not None:
+            payload["configuration"]["expectStatus"] = expect_status
+            self.__patch_a_synthetic_test(self.test_id, json.dumps(payload))
+        else:
+            print("expectStatus should not be none")
+
+    def patch_validation_string(self, validation_string):
+        """update validation string"""
+        payload = {"configuration": {"validationString": ""}}
+        if validation_string is None or validation_string == "":
+            print("validation string should not be none")
+        else:
+            payload["configuration"]["validationString"] = validation_string
+            self.__patch_a_synthetic_test(self.test_id, json.dumps(payload))
+
+    def patch_bundle(self, bundle):
+        """update bundle"""
+        payload = {
+            "configuration":
+                {
+                    "syntheticType": "HTTPScript",
+                    "scripts": {
+                        "bundle": "",  # required
+                        "scriptFile": ""  # required
+                    }
+                }
+        }
+        if bundle.endswith('.zip'):
+            with open(bundle, 'rb') as file1:
+                zip_content_byte = file1.read()
+                zip_content_byte_base64 = b64encode(zip_content_byte)
+                payload["configuration"]["scripts"]["bundle"] = zip_content_byte_base64.decode('utf-8')
+        else:
+            payload["configuration"]["scripts"]["bundle"] = bundle
+
+        self.__patch_a_synthetic_test(self.test_id, json.dumps(payload))
```

Comment:
@swethalohith2 There is an issue here, it is same as patch `--entry-file`, both bundle and scriptFile are required when patch a bundle test, here scriptFile is an empty dir, so after patch bundle test, scriptFile is also updated to empty string. Need to retrieve the test and fill correct scriptFile before patching the bundle test.